node/thermostat/main.js

```diff
@@ -0,0 +1,109 @@
+'use strict'
+
+const os = require('os')
+
+const DigitalTwinClient = require('azure-iot-digitaltwins-device').DigitalTwinClient
+
+const DeviceInformation = require('./deviceInformation').DeviceInformation
+const SdkInformation = require('./sdkInformation').SdkInformation
+const TemperatureSensor = require('./temperatureSensor').TemperatureSensor
+const Diagnostics = require('./diagnostics').Diagnostics
+
+const modelId = 'dtmi:com:example:Thermostat;1'
+const connectionString = process.env.DEVICE_CONNECTION_STRING
+
+let currentTemp = 20
+let targetTemp = 20
+let telemetryLoop = {}
+
+const propertyUpdateHandler = async (component, propertyName, reportedValue, desiredValue, version) => {
+  console.log('Received an update for ' + propertyName + ': ' + JSON.stringify(desiredValue))
+  targetTemp = parseFloat(JSON.stringify(desiredValue))
+  await adjustTemp(targetTemp)
+  await digitalTwinClient.report(component, { [propertyName]: desiredValue }, {
+    code: 200,
+    description: 'success',
+    version: version
+  })
+  console.log('updated the property')
+}
+
+const commandHandler = async (request, response) => {
+  console.log('received command: ' + request.commandName + ' for component: ' + request.componentName + ' with payload ' + request.payload)
+  if (request.commandName === 'reboot') {
+    // currentTemp = 0
+    targetTemp = 21
+    const delay = parseInt(request.payload)
+    clearInterval(telemetryLoop)
+    for (let index = 0; index < delay; index++) {
+      console.log('==================> REBOOT COMMAND RECEIVED <=================')
+      await sleep(index * 1000)
+    }
+    await startTelemetryLoop()
+    response.acknowledge(200, 'Command processed successfully')
+    console.log('acknowledgement succeeded.')
+  } else {
+    response.acknowledge(404, `Unknown command: ${request.commandName}`)
+  }
+}
+
+const digitalTwinClient = DigitalTwinClient.fromConnectionString(modelId, connectionString)
+
+const tempSensor = new TemperatureSensor('tempSensor1', propertyUpdateHandler)
+const deviceInformation = new DeviceInformation('deviceInfo')
+const sdkInformation = new SdkInformation('sdkInfo')
+const diag = new Diagnostics('diag', commandHandler)
+
+const thisSdkInfo = {
+  language: 'node',
+  version: '1.0.0-pnp-refresh.0',
+  vendor: 'Microsoft'
+}
+
+const thisDeviceInfo = {
+  manufacturer: 'Azure IoT Samples',
+  model: 'Thermostat.PnP-PPR-Node',
+  swVersion: '0.1',
+  osName: os.platform(),
+  processorArchitecture: os.arch(),
+  processorManufacturer: 'Contoso Industries',
+  totalStorage: 65000,
+  totalMemory: os.totalmem()
+}
+
+const sleep = (ms) => {
+  return new Promise(resolve => setTimeout(resolve, ms))
+}
+
+const adjustTemp = async (target) => {
+  const step = (parseFloat(target) - currentTemp) / 10
```

Comment:
>target [](start = 27, length = 6)

Wasn't target already a float?

Reply:
good catch, I guess the `parseFloat` is not needed. Updating